.devcontainer/postCreateCommand.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+export DEBIAN_FRONTEND=noninteractive
+sudo apt-get -y --force-yes update
+sudo apt-get -y --force-yes install libgdal-dev
+echo CPLUS_INCLUDE_PATH=/usr/include/gdal >> ~/.bashrc
+echo C_INCLUDE_PATH=/usr/include/gdal >> ~/.bashrc
+pip install --upgrade pip
+pip install --user -r ./requirements/datasets.txt
+pip install --user -r ./requirements/docs.txt
+pip install --user -r ./requirements/required.txt
+pip install --user -r ./requirements/style.txt
+pip install --user -r ./requirements/tests.txt
```

Comment:
These files are not intended to be used outside of CI. Would `pip install .[datasets,docs,style,tests]` work? That might install TorchGeo itself though, not sure if that would be problematic. There is also `pip install -e`.

Reply:
As it's a devcontainer, `docs` `style` and `tests` requirements are usefull.
The goal of the devcontainer is move some CI post commit processing (such as styling and test) on the dev IDE side.
Feature like format on save are quite usefull and deploy by default in the devcontainer

Reply:
Yes, and the pip install step listed above will also install all of those optional dependencies. See https://torchgeo.readthedocs.io/en/stable/user/installation.html

Reply:
Yes the purpose of the container is to ship a ready development container without the package.

In order to use a `pip install -e .` we will need to copy the whole torchgeo project in the container which is not great.

The workflow is:
1) The editor will launch the devcontainer image with only Python + public requirements.
2) The editor will then bind the torchgeo source code into the container.
3) Maybe add in `postStartCommand ` a `pip install -e .` that will run after the start of the devcontainer and install the current torchgeo lib.

At the end the devcontainer docker image will not change much over the time (only when dependencies upgrade).

Reply:
> we will need to copy the whole torchgeo project in the container which is not great.

> The editor will then bind the torchgeo source code into the container.

How are these different?

Reply:
Oups my bad, your right !!
The `pip install` must be done at the Dockerfile level (which is not specified here).
As you said the nbmake example is pretty good

This way of doing will work but is quite suboptimal.

I'm going to refactor this PR, in something more simple and robust